src/main/java/munch/TaskList.java

```diff
@@ -0,0 +1,142 @@
+package munch;
+
+import AddTasks.Deadlines;
+import AddTasks.Events;
+import AddTasks.Task;
+import AddTasks.Todo;
+import Exceptions.IncompleteInputException;
+import Exceptions.InvalidInputException;
+import munch.Parser;
+
+import java.lang.reflect.Array;
+import java.time.LocalDate;
+import java.time.format.DateTimeParseException;
+import java.util.ArrayList;
+
+public class TaskList {
+
+    public static void run(ArrayList<Task> tasks, String word, String[] words) throws IncompleteInputException, InvalidInputException {
```

Comment:
Perhaps this method is a little too complicated to be in TaskList? Here, the TaskList is performing the job of parsing input, interpreting the command and running the command

An abstract command class could be created for different commands to inherit from, for them to handle their own execution